bin/dist_get

```diff
@@ -0,0 +1,143 @@
+#!/bin/bash
+
+die() {
+	echo "$@" >&2
+	exit 1
+}
+
+have_binary() {
+	type "$1" > /dev/null
+}
+
+check_writeable() {
+	printf "" > "$1" && rm "$1"
+}
+
+download() {
+	local url="$1"
+	local output="$2"
+
+	if [ -z "$url" ] || [ -z "$output" ]; then
+		die "download takes exactly two arguments. was given '$@'"
+	fi
+
+	if ! check_writeable "$output"; then
+		die "download error: cannot write to $output"
+	fi
+
+	printf 'Downloading "%s" to "%s"\n' "$url" "$output"
+	if have_binary wget; then
+		wget "$url" -O "$output"
+	elif have_binary curl; then
+		curl --silent "$url" > "$output"
+	elif have_binary fetch; then
+		fetch "$url" -o "$output"
+	else
+		die "no binary found to download $url. exiting."
+	fi
+}
```

Comment:
It is dangerous to not immediately check if the download was successful.
For example if someone adds an "echo 'end of download'" just before the end of the function, then the exit code of the download instruction will not be the exit code of the download() function anymore.